update.sh

```diff
@@ -9,46 +11,65 @@ if [ ${#versions[@]} -eq 0 ]; then
 fi
 versions=( "${versions[@]%/}" )
 
-
-template=
-dockerfile=
+# Global variables
+arch=$(get_arch)
 
 yarnVersion="$(curl -sSL --compressed https://yarnpkg.com/latest-version)"
 
 function update_node_version {
+
+	local template=$1
+	shift
+        local dockerfile=$1
+        shift
+        local variant=
+        if [[ $# -eq 1 ]]; then
+                variant=$1
+                shift
+        fi
+
 	fullVersion="$(curl -sSL --compressed 'https://nodejs.org/dist' | grep '<a href="v'"$version." | sed -E 's!.*<a href="v([^"/]+)/?".*!\1!' | cut -f 3 -d . | sort -n | tail -1)"
 	(
 		cp $template $dockerfile
-		sed -E -i.bak 's/^(ENV NODE_VERSION |FROM node:).*/\1'"$version.$fullVersion"'/' "$dockerfile"
-		rm "$dockerfile.bak"
-		sed -E -i.bak 's/^(ENV YARN_VERSION ).*/\1'"$yarnVersion"'/' "$dockerfile"
-		rm "$dockerfile.bak"
-		if [[ "${version/.*/}" -ge 8 ]]; then
-			sed -E -i.bak 's/FROM alpine:3.4/FROM alpine:3.6/' "$dockerfile"
+		local fromprefix=
+		if [[ "$arch" != "x64" && "$variant" != "onbuild" ]]; then
+			fromprefix="$arch\/"
+		fi
+
+		sed -E -i.bak 's/^FROM (.*)/FROM '"$fromprefix"'\1/' "$dockerfile" && rm "$dockerfile".bak
+		sed -E -i.bak 's/^(ENV ARCH ).*/\1'"$arch"'/' "$dockerfile" && rm "$dockerfile".bak
+		sed -E -i.bak 's/^(ENV NODE_VERSION |FROM .*node:).*/\1'"$version.$fullVersion"'/' "$dockerfile" && rm "$dockerfile".bak
+		sed -E -i.bak 's/^(ENV YARN_VERSION ).*/\1'"$yarnVersion"'/' "$dockerfile" && rm "$dockerfile".bak
+		if [[ "${version/.*/}" -ge 8 || "$arch" = "ppc64le" ]]; then
+			sed -E -i.bak 's/FROM (.*)alpine:3.4/FROM \1alpine:3.6/' "$dockerfile"
 			rm "$dockerfile.bak"
 		fi
+
+		if [[ "$variant" = "alpine" ]]; then
+		        if [[ "$arch" = "ppc64le" ]]; then
+			        # Only alpine on ppc64 need 'COPY' for patch files
+			        sed -E -i.bak 's/^COPY patches/COPY patches\/ppc64le-v8patch.sh \/patches.sh/' "$dockerfile"
+			        rm "$dockerfile.bak"
+                        else
+			        sed -E -i.bak 's/^COPY patches//' "$dockerfile"
+			        rm "$dockerfile.bak"
+                        fi
+                fi
 	)
 }
 
 for version in "${versions[@]}"; do
 	# Skip "docs" and other non-docker directories
 	[ -f "$version/Dockerfile" ] || continue
 
-	template="Dockerfile.template"
-	dockerfile="$version/Dockerfile"
-
-	update_node_version
+	update_node_version "Dockerfile.template" "$version/Dockerfile"
 
-	variants=$(echo "$version"/*/ | xargs -n1 basename)
+        variants=$(get_variants)
```

Comment:
hmm, seems the update.sh is using tab for indent and I am using space. let me scan through all of my changes and use tab. Thanks